db/crud.go

```diff
@@ -666,16 +663,28 @@ func (db *Database) Put(docid string, body Body) (newRevID string, doc *Document
 		return "", nil, base.HTTPErrorf(http.StatusBadRequest, "Invalid revision ID")
 	}
 	generation++
+
 	deleted, _ := body[BodyDeleted].(bool)
```

Comment:
Does this need to be removed after extraction?

Reply:
No. Looks like it does require this to be returned in the body. At the bottom of the function we add in:

```
body[BodyId] = docid
body[BodyRev] = newRevID
```

Reply:
I can't see anywhere we're using the returned doc.Body